src/main/java/pl/psnc/dei/ui/components/imports/CreateImportComponent.java

```diff
@@ -0,0 +1,256 @@
+	private Import anImport;
+
+	private HorizontalLayout switchingTables;
+	private Component actionButtons;
+
+	private Input importName;
+	private Project project;
+
+	public CreateImportComponent(ImportPackageService importPackageService, RecordsRepository recordsRepository, ProjectsRepository projectsRepository) {
+		this.importPackageService = importPackageService;
+		this.recordsRepository = recordsRepository;
+		this.allRecords = new HashSet<>();
+		this.projectsRepository = projectsRepository;
+		this.selectedRecordsForImport = new HashSet<>();
+		createComponent();
+	}
+
+	public CreateImportComponent(ImportPackageService importPackageService, Import anImport, RecordsRepository recordsRepository, ProjectsRepository projectsRepository) {
+		this.anImport = anImport;
+		this.importPackageService = importPackageService;
+		this.recordsRepository = recordsRepository;
+		this.projectsRepository = projectsRepository;
+		this.selectedRecordsForImport = recordsRepository.findAllByAnImport(anImport);
+		if (!selectedRecordsForImport.isEmpty()) {
+			Project project = selectedRecordsForImport.iterator().next().getProject();
+			this.allRecords = recordsRepository.findAllByProjectAndAnImportNull(project);
+		} else {
+			this.projectsRepository = projectsRepository;
+			this.selectedRecordsForImport = new HashSet<>();
+			this.allRecords = new HashSet<>();
+			this.anImport = null;
+		}
+		createComponent();
+	}
+
+	private Component createProjectSelection() {
+		Select<Project> projectSelect = new Select<>();
+		ListDataProvider<Project> listDataProvider = new ListDataProvider<>(projectsRepository.findAll());
+		projectSelect.setDataProvider(listDataProvider);
+
+		projectSelect.setEnabled(anImport == null);
+		projectSelect.addValueChangeListener(event -> {
+			Project project = (Project) event.getValue();
+			this.project = project;
+			allRecords = recordsRepository.findAllByProjectAndAnImportNull(project);
+			selectedRecordsForImport = new HashSet<>();
+			refresh();
+		});
+		HorizontalLayout projectSelectionLayout = new HorizontalLayout();
+		projectSelectionLayout.add(new Label("Select project"));
+		projectSelectionLayout.add(projectSelect);
+		return projectSelectionLayout;
+	}
+
+	private void refresh() {
+		if (switchingTables != null) {
+			remove(switchingTables);
+		}
+		if (actionButtons != null) {
+			remove(actionButtons);
+		}
+		switchingTables = new HorizontalLayout();
+		allRecordsGrid = generateRecordsGrid(allRecords);
+		switchingTables.add(allRecordsGrid);
+		switchingTables.add(generateSwitchingButtons());
+		selectedRecordsGrid = generateRecordsGrid(selectedRecordsForImport);
+		switchingTables.add(selectedRecordsGrid);
+		add(switchingTables);
+		actionButtons = generateActionButtons();
+		add(actionButtons);
+	}
+
+	private void createComponent() {
+		if (anImport == null) {
+			add(createProjectSelection());
+			HorizontalLayout importNameLayout = new HorizontalLayout();
+			importName = new Input();
```

Comment:
When this is a new import I think we should give it a default name with possibility to change it.